src/vs/editor/contrib/suggest/suggestWidget.ts

```diff
@@ -521,7 +513,22 @@ export class SuggestWidget implements IContentWidget, IListVirtualDelegate<Compl
 		this.list = new List('SuggestWidget', this.listElement, this, [renderer], {
 			useShadows: false,
 			openController: { shouldOpen: () => false },
-			mouseSupport: false
+			mouseSupport: false,
+			accessibilityProvider: {
+				getAriaLabel: (item: CompletionItem) => {
+					if (this.expandDocsSettingFromStorage()) {
+						const { documentation, detail } = item.completion;
+						const docs = strings.format(
```

Comment:
This is tricky... Usually, detail and documentation aren't defined and the UI triggers that when an item gains focus. I don't think tho that this call is sync'd on that and what you get is likely indeterministic. You should be able to call `resolve` on the item but you need to know how long you can "wait" for its result...

Reply:
Yes, I missed this. 
However this is just working since once the suggestion details and docs get resolved the item will get re-rendered by the `splice` call [here](https://github.com/Microsoft/vscode/blob/21731345de457a1e3463b16283ba29ae799d9f62/src/vs/editor/contrib/suggest/suggestWidget.ts#L692).
When the item gets rerendered we will return deterministicly the correct aria label since the item already got resolved.

I can not wait in the `getAriaLabel` call since it is not async.

Idealy the method would also check if the item is resolved, if not just return the `label`. However not sure how to check that.
I am open for suggestions.

Reply:
Yeah, would just be nicer if we wouldn't read it twice. To know if its resolved, you can add a boolean to the `CompletionItem` which knows that resolving is done.

Reply:
As you suggested I have added a `isResolved` boolean to the `CompletionItem`. So when we compute the aria label if the item is not resolved we will only read the `label`.
Let me know what you think.

Reply:
Yeah, looking better now